src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1202,7 +1202,7 @@ public Image getIcon(int width, int height) {
     static Image getSystemIcon(SystemIcon iconType) {
         long hIcon = getSystemIcon(iconType.getIconID());
```

Comment:
What is hIcon value here? If it is 0 we can simply add the same condition as in the getShell32Icon and avoid all the makeIcon and disposeIcon calls. Or we have a valid hIcon code but makeIcon is unable to construct an icon for it?

Reply:
Yes, I think you should add the condition `if (hIcon != 0)` before calling `makeIcon(hIcon)`.

Reply:
Updated